.github/workflows/build-deploy.yml

```diff
@@ -0,0 +1,85 @@
+name: "Build, push to GHCR and deploy image to GKE cluster"
+
+on:
+  push:
+    tags: [ '*' ]
+
+env:
+  RELEASE_TAG: ${{ github.ref_name }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+        id: buildx
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          builder: ${{ steps.buildx.outputs.name }}
+          platforms: linux/amd64,linux/arm64 # Remember image in dockerfile should also support these platforms
```

Comment:
I can't take credit for this, since it was already present :D
It's presented like this because I renamed the file to something more related.
What I added: 
env vars, tagging, helm, google auth.